core/io/resource_loader.cpp

```diff
@@ -823,12 +823,10 @@ String ResourceLoader::_path_remap(const String &p_path, bool *r_translation_rem
 
 	if (path_remaps.has(new_path)) {
 		new_path = path_remaps[new_path];
-	}
-
-	if (new_path == p_path) { // Did not remap.
```

Comment:
The thing here is that new_path may have been changed earlier in the function due to a translation remap.